### PR TITLE
CrayPE fixes for MAGMA

### PR DESCRIPTION
**Pull request by lukebroskop**
Force CCE to compile fortran modules with lower case names
Allow for CCE to be the host compuiler with nvcc

### CODE REVIEW

**Review by eugeneswalker — DISMISSED** [review]

File: var/spack/repos/builtin/packages/magma/package.py [hunk 1]
```diff
@@ -74,6 +74,9 @@ def cmake_args(self):
         options += ['-DBUILD_SHARED_LIBS=%s' %
                     ('ON' if ('+shared' in spec) else 'OFF')]
 
+        if spec.satisfies('%cce'):
+            options += [('-DCUDA_NVCC_FLAGS=-allow-unsupported-compiler')]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/magma/package.py]: Not strictly necessary, but eliminates unnecessary parens:

```suggestion
            options += ['-DCUDA_NVCC_FLAGS=-allow-unsupported-compiler']
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
See suggestion.

File: var/spack/repos/builtin/packages/magma/package.py (hunk 1 above)
Comment (lukebroskop) [reply var/spack/repos/builtin/packages/magma/package.py]: Fixed, thanks for reviewing.

**Comment by tldahlgren** [comment]
Not a consistent way of adding to the options -- `+=`, `extends`, and `append` -- but willing to differ that issue to a separate PR since the changes here aren't the only place where inconsistency has been introduced.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**